compose/project.py

```diff
@@ -508,7 +510,13 @@ def _get_convergence_plans(self, services, strategy):
                 log.debug('%s has upstream changes (%s)',
                           service.name,
                           ", ".join(updated_dependencies))
-                plan = service.convergence_plan(ConvergenceStrategy.always)
+                containers_stopped = any((not c.is_paused and not c.is_restarting and not c.is_running
```

Comment:
I think you can do that more efficiently with `service.containers(stopped=True, filters={'status': ['created', 'exited']})`. See https://docs.docker.com/engine/api/v1.30/#operation/ContainerList for the filters docs